### ISSUE TITLE
Image.save() doesn't work when saving more than 10 images at a time

### ISSUE THREAD

**Issue by ks233** [processing/p5.js#6021]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [X] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

1.5.0

### Web browser and version

Edge 110.0.1587.46

### Operating System

Windows 10

### Steps to reproduce this

When the number of images saved at one time exceeds 10, `Image.save()` will not work.

For example, if I run the following code, only images 0.jpg to 9.jpg will be saved, and images 10 to 14 will not be saved:

```js
function test(){
	for(i = 0;i<15;i++){
		img.save(`${i}.jpg`);
	}
}
```

I am running the program on VS Code Live Server using the MS Edge browser. Is this a bug or a browser issue?

**Comment by davepagurek** [comment]
I've run into something similar in Chrome, it seems like Chrome just loses files if you save too often. Try adding a bit of a delay between each save, e.g.:

```js
function delay(ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}

async function test() {
  for(let i = 0; i < 15; i++) {
    img.save(`${i}.jpg`)
    await delay(250)
  }
}
```

...or if you're exporting once per frame, try setting `frameRate(4)` or something else low.

**Comment by Tapo41** [comment]
I want to work on this .

**Comment by davepagurek** [comment]
Thanks for your interest @Tapo41! before jumping into implementing a solution, we should probably research first what our options are for working around this. If you have an idea mind already, would you mind describing it a bit? If not, if you first want to spend some time testing out what approaches might be feasible, that would also be greatly appreciated! And then after we've discussed in this issue here, we can start an implementation.

**Comment by somendrakr** [comment]
hello @davepagurek  I have checked this issue and its not due to a bug in p5.js library. Its happens with all the programs and is related to the operating system you are using In each system there is a limitation on number of simultaneous downloads from the browser at a time for windows it seems to be 9. There are some ways to fixed it one that you have already mentioned by using delay other is doing edit in registry editor of windows and increasing the download limit. Although I was not getting desired result in second method but its mentioned in many blogs on internet.

**Comment by davepagurek** [comment]
Thanks @somendrakr! If we think this is a common enough problem that we want to solve it, we could potentially introduce a "download queue" of sorts that keeps the images in memory but that only saves images out of the queue as a download at a low rate to prevent too many concurrent downloads. That said, it would probably be faster to use something like [ccapture](https://github.com/spite/ccapture.js) to mix and download one output file if one is exporting images for an animation, and since this isn't an issue unique to p5, it's probably also fine if we don't try to solve this in p5. Maybe this would best be solved as a library for p5 first, if there's interest? @limzykenneth does that sound like the right call to you too?

**Comment by limzykenneth** [comment]
Yeah, as @davepagurek alluded to, for consecutive saving of each frames, the best way to do it is with something like ccapture that is exclusively made for that kind of use case. It won't be very suitable for p5.js to implement it in the core library because things like ccapture does quite a lot more than queueing downloads (eg. it also changes how the animation loop is called as well).

**Comment by sleep9** [comment]
@limzykenneth Do you have any clue how to use ccapture with processing properly  with and without instance mode? It freezes the animation loop after calling capture and calling `window.requestAnimationFrame(p5.draw)` glitches out...

I am running a custom fork of p5.js 1.11.7 but no idea where to start to get this working again. Any tips?

**Comment by limzykenneth** [comment]
@sleep9 I'll have a look but it'll take a few days as I'm still working through other work stuff.

@ksen0 Just pinging here as well since I see you have replied in #4481

**Comment by sleep9** [comment]
Not a clutch issue. upgrading to 1.9.0 worked with ccapture and then I ended up not using ccapture in the end.

**Comment by limzykenneth** [comment]
Alright, after quite a bit of digging I finally found out why it's not working. The change all the way back in https://github.com/processing/p5.js/commit/abe5c95ff1a20821bb9743a5ef18e002787c9bfb, specifically the use of `window.performance.now()` makes the animation loop no longer progress when using CCapture because CCapture also overwrites the functionality of `window.performance.now()`. 

The overall setup and interaction between CCapture and p5.js' animation loop is too complex to properly disentangle for me. CCapture also does not seem to be maintained for several years now. I'm considering alternative that is simpler than CCapture, @sleep9 what did you end up using? There might be a case for an implementation with p5.js 2.x addon with post draw hook and web worker, which is just an idea I have off the top of my head right now.

**Comment by limzykenneth** [comment]
I've made [p5.record.js](https://github.com/limzykenneth/p5.record.js) that uses native canvas capture API which can help capture videos of canvas including interactive sketches which is more difficult with ccapture. Possible next plan is to implement image sequence capture which can fully resolve this issue but for video recording it should work well now with p5.record.js.

Edit: [p5.record.js](https://github.com/limzykenneth/p5.record.js) now also support capturing image sequence, I'll close this issue for now.
